agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/Configuration.java

```diff
@@ -579,6 +584,67 @@ public void validate() throws FriendlyException {
         public GcReportingLevel reportingLevel = GcReportingLevel.TENURED_ONLY;
     }
 
+    public static class AadAuthentication {
+        public boolean enabled;
+        public AuthenticationType type;
+        public String clientId;
+        public String keePassDatabasePath;
+        public String tenantId;
+        public String clientSecret;
+        public String authorityHost;
+        public AadAuthentication() {
+            this.enabled = false;
+        }
+        
+        public boolean validate() throws FriendlyException {
```

Comment:
this method never returns false
```suggestion
        public void validate() throws FriendlyException {
```